source/renderer/app/components/settings/categories/StakePoolsSettings.js

```diff
@@ -0,0 +1,171 @@
+// @flow
+import React, { Component } from 'react';
+import { map } from 'lodash';
+import { Select } from 'react-polymorph/lib/components/Select';
+import { SelectSkin } from 'react-polymorph/lib/skins/simple/SelectSkin';
+import { observer } from 'mobx-react';
+import { defineMessages, intlShape } from 'react-intl';
+import { isValidUrl } from '../../../utils/validations';
+import { getSmashServerIdFromUrl } from '../../../utils/staking';
+import InlineEditingInput from '../../widgets/forms/InlineEditingInput';
+import styles from './StakePoolsSettings.scss';
+import {
+  SMASH_SERVERS_LIST,
+  SMASH_SERVER_TYPES,
+} from '../../../config/stakingConfig';
+import type { SmashServerType } from '../../../types/stakingTypes';
+
+import LocalizableError from '../../../i18n/LocalizableError';
+
+const messages = defineMessages({
+  smashSelectLabel: {
+    id: 'settings.stakePools.smash.select.label',
+    defaultMessage: '!!!Off-chain metadata server (SMASH)',
+    description:
+      'smashSelectLabel for the "Smash" selection on the Stake Pools settings page.',
+  },
+  smashSelectCustomServer: {
+    id: 'settings.stakePools.smash.select.placeholder',
+    defaultMessage: '!!!Custom server',
+    description:
+      'smashSelectCustomServer option for the "Smash" selection on the Stake Pools settings page.',
+  },
+  smashURLInputLabel: {
+    id: 'settings.stakePools.smashUrl.input.label',
+    defaultMessage: '!!!SMASH server URL',
+    description:
+      'smashURLInputLabel for the "Smash Custom Server" selection on the Stake Pools settings page.',
+  },
+  smashUrlInputPlaceholder: {
+    id: 'settings.stakePools.smashUrl.input.placeholder',
+    defaultMessage: '!!!Enter custom server URL',
+    description:
+      'smashUrlInputPlaceholder for the "Smash Custom Server" selection on the Stake Pools settings page.',
+  },
+  smashUrlInputInvalidUrl: {
+    id: 'settings.stakePools.smashUrl.input.invalidUrl',
+    defaultMessage: '!!!Invalid URL',
+    description:
+      'smashUrlInputInvalidUrl for the "Smash Custom Server" selection on the Stake Pools settings page.',
+  },
+});
+
+type Props = {
+  smashServerUrl: string,
+  smashServerUrlError?: ?LocalizableError,
+  onSelectSmashServerUrl: Function,
+  onResetSmashServerError: Function,
+  isLoading: boolean,
+};
+
+type State = {
+  editingSmashServerUrl: string,
+  successfullyUpdated: boolean,
+  wasLoading: boolean,
```

Comment:
Is there any special reason of `wasLoading` state variable usage?
Code shows it just tracks `isLoading` prop and same value always.

Reply:
@yakovkaravelov `isLoading` has not the same value always. You can check `smashServerLoading` inside `StakingStore.js`. It's an easy and reliable way to check if the success message should be displayed.

Reply:
@daniloprates I meant `wasLoading` is always same value as `isLoading`, so I was curious using `isLoading` prop directly instead of using `wasLoading` state would be ideal or not.